### DISCUSSION TITLE
How could I put the admin code in a custom Python file named admin.py instead of the file where the FastAPI instance is located?

### DISCUSSION THREAD

**Discussion by ChengChe106** [aminalaee/sqladmin#749] [Q&A] [answered]
I put the admin code in a custom Python file named admin.py, but when I try to access http://127.0.0.1:8080/admin/, it shows "Not Found".

This is my code.
project/src/admin.py

from sqladmin import Admin, ModelView

from src import app
from src.database import engine
from src.user.model import User

admin = Admin(app, engine)

class UserAdmin(ModelView, model=User):
    column_list = [User.id, User.username, User.is_active, User.is_superuser]

admin.add_view(UserAdmin)

**Comment by hasansezertasan** [comment]
I think the solution for this could be a more like a Flask Admin approach, meaning:

```python
...
admin = Admin()

app =...

admin.init_app(app)
...
```

**Reply by aminalaee** [reply]
Yes I agree, I haven't had a chance to work on this, but this is definitely a good approach.

**Reply by ChengChe106** [reply]
Thank you for your reply. I look forward to the feature becoming available.

**Reply by im-redactd** [reply, marked as answer]
You can do this now and it's fairly easy.

folder structure for my example is
```
app/
- admin/
-- setup.py
-- views/
--- user_admin.py
main.py
```

In your main.py 

```
from app.admin.setup import init_admin
admin = init_admin(app)

```

your app/admin/setup.py would look like this
```
from sqladmin import Admin
from fastapi import FastAPI
from src.database import engine
from app.admin.views.user_admin import UserAdmin

def init_admin(app: FastAPI):
    admin = Admin(app, engine, title="App Title", templates_dir="app/templates")
    admin.add_view(UserAdmin)
    return admin
```
your app/admin/views/user_admin.py - you may not need to go to this level of organization if you don't have a lot of views and you could add them to the setup.py but I wanted to include this option that I use.

```
from sqladmin import ModelView
from src.user.model import User

class UserAdmin(ModelView, model=User):
    column_list = [User.id, User.username, User.is_active, User.is_superuser]
 
 ```

**Reply by ChengChe106** [reply]
Thank you! This is helpful! I believe this is the best solution until the author updates the library.

**Reply by aminalaee** [reply]
Feel free to submit a PR to add this functionality
